lib/zuora/rest/client.rb

```diff
@@ -84,32 +85,68 @@ def handle_rate_limiting(method, url, params = nil)
         end
       end
 
+
       # @param [Faraday::Response] response
       # @throw [ErrorResponse] if unsuccessful
       # @return [Faraday::Response]
       def fail_or_response(response)
         if response.status != 200
           fail(ErrorResponse.new("HTTP Status #{response.status}", response))
-        elsif !response.body['success']
-          errors = 'Not successful.'
-
-          if response.body['reasons']
-            reasons = response.body['reasons'].map do |reason|
-              "Error #{reason['code']}: #{reason['message']}"
-            end
-            errors += ' ' + reasons.join(', ')
+        elsif response.body.kind_of?(Array)
```

Comment:
Does it matter if it is an Array? Could this be `response.body.respond_to?(:first)`

Reply:
```
irb(main):001:0> {}.respond_to?(:first)
=> true
irb(main):002:0> [].respond_to?(:first)
=> true
```